developer/14.0/reference/file-types/model-ts.php

```diff
@@ -127,35 +133,37 @@ function</dfn> is responsible for. It is a little bit of code that looks at some
 specify one: </p>
 
 <pre><code class="lang-typescript">searchTermToKey: function (term) {
-  // Use this pattern to remove common diacritical marks.
+  // Use this pattern to remove common diacritical marks (accents).
   // See: https://www.compart.com/en/unicode/block/U+0300
   const COMBINING_DIACRITICAL_MARKS = /[\u0300-\u036f]/g;
 
-  // Converts to Unicode Normalization form D.
-  // This means that MOST accents and diacritics have been "decomposed" and
-  // are stored as separate characters. We can then remove these separate
-  // characters!
-  //
-  // e.g., Å → A + ˚
-  let normalizedTerm = term.normalize('NFD');
+  // Lowercase each letter in the string INDIVIDUALLY.
+  // Why individually? Some languages have context-sensitive lowercasing
+  // rules (e.g., Greek), which we would like to avoid.
+  // So we convert the string into an array of code points (Array.from(term)),
+  // convert each individual code point to lowercase (.map(c => c.toLowerCase())),
+  // and join the pieces back together again (.join(''))
+  let lowercasedTerm = Array.from(term).map(c => c.toLowerCase()).join('');
 
-  // Now, make it lowercase.
+  // Once it's lowercased, we convert it to NFKD normalization form
+  // This does many things, such as:
   //
-  // e.g.,  A + ˚ → a + ˚
-  let lowercasedTerm = normalizedTerm.toLowerCase();
-
-  // Now, using the pattern above replace each accent and diacritic with the
+  //  - separating characters from their accents/diacritics
+  //      e.g., "ï" -> "i" + "◌̈"
```

Comment:
This renders funny on some systems -- perhaps instead:
```suggestion
  //      e.g., "ï" -> "i" + "¨" (U+0308)
```